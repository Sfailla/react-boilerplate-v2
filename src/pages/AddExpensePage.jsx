import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import ExpenseForm from '../components/ExpenseForm';

export class AddExpensePage extends React.Component {
	onSubmit = expense => {
		this.props.onSubmit(expense);
		this.props.history.push('/');
	};

	render() {
		return (
			<Fragment>
				<h2>Add Expense Component</h2>
				<ExpenseForm onSubmit={this.onSubmit} />
			</Fragment>
		);
	}
}

const mapDispatchToProps = dispatch => {
	return {
		onSubmit: expense => dispatch(addExpense(expense))
	};
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
