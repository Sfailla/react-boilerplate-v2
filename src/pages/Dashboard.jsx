import React from 'react';

const ExpenseDashboardPage = () => <div>this is from my dashboard component</div>;

export default ExpenseDashboardPage;
