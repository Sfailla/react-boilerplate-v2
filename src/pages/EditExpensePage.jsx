import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import ExpenseForm from '../components/ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
	onSubmit = expense => {
		this.props.editExpense(this.props.expense.id, expense);
		this.props.history.push('/');
	};

	onRemove = () => {
		this.props.removeExpense({ id: this.props.expense.id });
		this.props.history.push('/');
	};

	render() {
		return (
			<Fragment>
				<ExpenseForm
					expense={this.props.expense}
					onSubmit={this.onSubmit}
				/>
				<button onClick={this.onRemove}>Remove</button>
			</Fragment>
		);
	}
}

const mapStateToProps = (state, props) => ({
	expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => {
	return {
		editExpense: (id, expense) =>
			dispatch(editExpense(props.expense.id, expense)),
		removeExpense: id => dispatch(removeExpense({ id }))
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
