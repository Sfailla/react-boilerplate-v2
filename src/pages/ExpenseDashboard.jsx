import React, { Fragment } from 'react';
import ExpenseList from '../components/ExpenseList';
import ExpenseListFilters from '../components/ExpenseListFilters';
import ExpensesSummary from '../components/ExpensesSummary';

const ExpenseDashboardPage = () => (
	<Fragment>
		<ExpensesSummary />
		<ExpenseList />
		<ExpenseListFilters />
	</Fragment>
);
export default ExpenseDashboardPage;
