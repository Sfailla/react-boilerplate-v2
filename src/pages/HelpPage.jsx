import React from 'react';

const HelpPage = () => <div>this is my help component</div>;

export default HelpPage;
