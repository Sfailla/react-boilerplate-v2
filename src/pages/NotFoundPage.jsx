import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = (props) => {
	console.log(props);
	return (
		<div>
			404! -- <Link to="/">Go back</Link>
		</div>
	);
};

export default NotFoundPage;
