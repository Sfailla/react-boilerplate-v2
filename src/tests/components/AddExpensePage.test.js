import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../pages/AddExpensePage';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach(() => {
	onSubmit = jest.fn();
	history = { push: jest.fn() };
	wrapper = shallow(<AddExpensePage onSubmit={onSubmit} history={history} />);
});

test('should render AddExpensePage correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
	wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
	expect(history.push).toHaveBeenCalledWith('/');
	expect(onSubmit).toHaveBeenCalledWith(expenses[1]);
});
