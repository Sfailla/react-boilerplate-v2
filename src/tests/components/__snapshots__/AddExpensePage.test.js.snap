// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensePage correctly 1`] = `
<Fragment>
  <h2>
    Add Expense Component
  </h2>
  <ExpenseForm
    onSubmit={[Function]}
  />
</Fragment>
`;
